src/main/java/org/cactoos/iterator/Sliced.java

```diff
@@ -52,22 +71,77 @@
      */
     public Sliced(final int start, final int count,
         final Iterator<T> iterator) {
-        this.iterator = new HeadOf<>(
-            count,
-            new Skipped<>(
-                start,
-                iterator
-            )
-        );
+        this(start, iterator, index -> index > start + count - 1);
+    }
+
+    /**
+     * Constructor.
+     * Constructs a head iterator with a number of heading elements
+     * @param iterator Decorated iterator
+     * @param count Maximum number of elements for resulted iterator
+     */
+    public Sliced(final Iterator<T> iterator, final int count) {
+        this(0, count, iterator);
+    }
+
+    /**
+     * Constructor.
+     * Constructs an iterator of start position and up to the end
+     * @param start Starting index
+     * @param iterator Decorated iterator
+     */
+    public Sliced(final int start, final Iterator<T> iterator) {
+        this(start, iterator, any -> !iterator.hasNext());
+    }
+
+    /**
+     * Constructor.
+     * @param start Starting index
+     * @param iterator Decorated iterator
+     * @param end Predicate that test whether iterating should stop
+     */
+    private Sliced(final int start, final Iterator<T> iterator,
+        final Predicate<Integer> end) {
```

Comment:
@fanifieiev in cactoos, it is preferred to keep the iterator as the last argument of the constructor

Reply:
@fanifieiev by the way, since it's internal to the class, you can use `IntPredicate` here.

Reply:
Done